components/remotes/remotes.js

```diff
@@ -92,9 +92,10 @@ class RemotesViewModel {
   updateRemotes() {
     return this.server.getPromise('/remotes', { path: this.repoPath() })
       .then(remotes => {
+        const outerThis = this;
         remotes = remotes.map(remote => ({
           name: remote,
-          changeRemote() { this.currentRemote(remote) }
+          changeRemote() { outerThis.currentRemote(remote) }
```

Comment:
This would be better at it is more ES6 compliant
And no need of outerThis

```suggestion
          changeRemote: () => { outerThis.currentRemote(remote) }
```